### ISSUE TITLE
Confusing classification of `..` in the `import/order` rule.

### ISSUE THREAD

**Issue by jonathanj** [benmosher/eslint-plugin-import#1797]
The relevant rules I have configured:

```
  import/no-useless-path-segments: error

  import/order:
    - error
    - newlines-between: always
      groups:
        - builtin
        - external
        - internal
        - parent
        - sibling
        - index
      pathGroups:
        - pattern: "app/**"
          group: internal
        - pattern: "@/**"
          group: internal
```

This code:

```
import {…} from '..'
import {…} from '../context'

import {…} from './types'
```

produces this following error:

```
error  `..` import should occur after import of `./types`             import/order
```

Which is suggesting that my imports look like this:

```
import {…} from '../context'

import {…} from './types'

import {…} from '..'
```

If I change the import to instead be `import {…} from '../'` then I get only the following error:

```
error  Useless path segments for "../", should be ".."  import/no-useless-path-segments
```

On one hand `..` is considered part of no order grouping (or `index`, I can't quite tell which), on the other hand `../` is considered a parent import, except with a "useless" segment.

I'm not sure if I'm misunderstood or missed some part of the documentation but I'm rather confused. I would like `..` to group with imports like `../context`, since it is equivalent to `../index`.

**Comment by ljharb** [comment]
`..` and `../` should both be considered a parent import, for sure.

I agree that either one should be treated as the same import group as `../context`, since indeed they're shorthand for `../index` (or `../$packageJSONMain`, etc).

**Comment by golopot** [comment]
This is fixed in https://github.com/benmosher/eslint-plugin-import/pull/1658 and released in 2.20.2.
